crates/dbsp/src/trace/mod.rs

```diff
@@ -489,6 +489,30 @@ where
     /// [`Batch::recede_to`].
     fn recede_to(&mut self, frontier: &Self::Time);
 
+    /// Returns elements from `self` that satisfy a predicate.
+    fn filter(&self, predicate: &dyn Fn(&Self::Key, &Self::Val) -> bool) -> Self
```

Comment:
I think this should be pub
We need more functions like this

Reply:
It's a trait method, so it's automatically `pub`.